### PR TITLE
[discovery-gce] Upgrade all Google dependencies

### PR DESCRIPTION
**Pull request by arteam**
Upgrade the dependencies in order to avoid them being flagged for CVEs.

### CODE REVIEW

File: plugins/discovery-gce/build.gradle [hunk 1]
```diff
@@ -7,20 +7,30 @@ esplugin {
 }
 
 versions << [
-  'google': '1.23.0'
+  'google'              : '1.41.4',
```
Comment (fcofdez) [inline plugins/discovery-gce/build.gradle:10]: Sorry for the delay on this, I'm not sure if these versions align with what the library expects? 
```
[INFO] \- com.google.apis:google-api-services-compute:jar:v1-rev235-1.25.0:compile
[INFO]    \- com.google.api-client:google-api-client:jar:1.25.0:compile
[INFO]       +- com.google.oauth-client:google-oauth-client:jar:1.25.0:compile
[INFO]       |  +- com.google.http-client:google-http-client:jar:1.25.0:compile
[INFO]       |  |  +- org.apache.httpcomponents:httpclient:jar:4.5.5:compile
[INFO]       |  |  |  +- org.apache.httpcomponents:httpcore:jar:4.4.9:compile
[INFO]       |  |  |  +- commons-logging:commons-logging:jar:1.2:compile
[INFO]       |  |  |  \- commons-codec:commons-codec:jar:1.10:compile
[INFO]       |  |  \- com.google.j2objc:j2objc-annotations:jar:1.1:compile
[INFO]       |  \- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO]       +- com.google.http-client:google-http-client-jackson2:jar:1.25.0:compile
[INFO]       |  \- com.fasterxml.jackson.core:jackson-core:jar:2.9.6:compile
[INFO]       \- com.google.guava:guava:jar:20.0:compile
```
Am I missing something? @arteam

Comment (arteam) [reply plugins/discovery-gce/build.gradle:10]: You're correct! Let me pinpoint the exact versions of the transitive dependencies instead of the latest.

Comment (fcofdez) [reply plugins/discovery-gce/build.gradle:10]: It seems like they discontinued that library in favor of https://mvnrepository.com/artifact/com.google.cloud/google-cloud-compute/1.8.1?

Comment (arteam) [reply plugins/discovery-gce/build.gradle]: Interesting! Let me try this dependency

Comment (arteam) [reply plugins/discovery-gce/build.gradle]: It seems it still gets updated! https://repo1.maven.org/maven2/com/google/apis/google-api-services-compute/v1-rev20220322-1.32.1/ got released a couple of days go. It seems that the artefacts stopped being indexed by mvnrepository.com at some point. I should have used https://search.maven.org/search?q=google-api-services-compute from the beggining :facepalm:

Comment (arteam) [reply plugins/discovery-gce/build.gradle]: Okay, it's there https://mvnrepository.com/artifact/com.google.apis/google-api-services-compute/v1-rev20220322-1.32.1, just the sorting on https://mvnrepository.com/artifact/com.google.apis/google-api-services-compute is broken

Comment (fcofdez) [reply plugins/discovery-gce/build.gradle]: Oh 🤦, I was looking into the google cloud documentation and I didn't manage to find a good release list... thanks for looking into this!

**Comment by arteam** [comment]
@elasticmachine update branch

File: plugins/discovery-gce/build.gradle [hunk 2]
```diff
@@ -7,20 +7,30 @@ esplugin {
 }
 
 versions << [
-  'google': '1.23.0'
+  'google'              : '1.41.1',
+  'google_api_client'   : '1.33.1',
+  'api_services_compute': 'v1-rev20220322-1.32.1',
+  'google_oauth_client' : '1.33.1',
```
Comment (fcofdez) [inline plugins/discovery-gce/build.gradle:13]: I think this expects `1.33.0` 
```
[INFO] \- com.google.apis:google-api-services-compute:jar:v1-rev20220322-1.32.1:compile
[INFO]    \- com.google.api-client:google-api-client:jar:1.33.1:compile
[INFO]       +- com.google.oauth-client:google-oauth-client:jar:1.33.0:compile <---
[INFO]       +- com.google.http-client:google-http-client-gson:jar:1.41.1:compile
[INFO]       |  \- com.google.code.gson:gson:jar:2.8.9:compile
[INFO]       +- com.google.guava:guava:jar:31.0.1-jre:compile
[INFO]       |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO]       |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO]       |  +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO]       |  +- org.checkerframework:checker-qual:jar:3.12.0:compile
[INFO]       |  +- com.google.errorprone:error_prone_annotations:jar:2.7.1:compile
[INFO]       |  \- com.google.j2objc:j2objc-annotations:jar:1.3:compile
[INFO]       +- com.google.http-client:google-http-client-apache-v2:jar:1.41.1:compile
[INFO]       +- org.apache.httpcomponents:httpcore:jar:4.4.15:compile
[INFO]       +- org.apache.httpcomponents:httpclient:jar:4.5.13:compile
[INFO]       |  +- commons-logging:commons-logging:jar:1.2:compile
[INFO]       |  \- commons-codec:commons-codec:jar:1.11:compile
[INFO]       \- com.google.http-client:google-http-client:jar:1.41.1:compile
[INFO]          +- io.opencensus:opencensus-api:jar:0.30.0:compile
[INFO]          |  \- io.grpc:grpc-context:jar:1.27.2:compile
[INFO]          \- io.opencensus:opencensus-contrib-http-util:jar:0.30.0:compile
```

Comment (fcofdez) [inline plugins/discovery-gce/build.gradle:13]: I think this expects `1.33.0` 
```
[INFO] \- com.google.apis:google-api-services-compute:jar:v1-rev20220322-1.32.1:compile
[INFO]    \- com.google.api-client:google-api-client:jar:1.33.1:compile
[INFO]       +- com.google.oauth-client:google-oauth-client:jar:1.33.0:compile <---
[INFO]       +- com.google.http-client:google-http-client-gson:jar:1.41.1:compile
[INFO]       |  \- com.google.code.gson:gson:jar:2.8.9:compile
[INFO]       +- com.google.guava:guava:jar:31.0.1-jre:compile
[INFO]       |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO]       |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO]       |  +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO]       |  +- org.checkerframework:checker-qual:jar:3.12.0:compile
[INFO]       |  +- com.google.errorprone:error_prone_annotations:jar:2.7.1:compile
[INFO]       |  \- com.google.j2objc:j2objc-annotations:jar:1.3:compile
[INFO]       +- com.google.http-client:google-http-client-apache-v2:jar:1.41.1:compile
[INFO]       +- org.apache.httpcomponents:httpcore:jar:4.4.15:compile
[INFO]       +- org.apache.httpcomponents:httpclient:jar:4.5.13:compile
[INFO]       |  +- commons-logging:commons-logging:jar:1.2:compile
[INFO]       |  \- commons-codec:commons-codec:jar:1.11:compile
[INFO]       \- com.google.http-client:google-http-client:jar:1.41.1:compile
[INFO]          +- io.opencensus:opencensus-api:jar:0.30.0:compile
[INFO]          |  \- io.grpc:grpc-context:jar:1.27.2:compile
[INFO]          \- io.opencensus:opencensus-contrib-http-util:jar:0.30.0:compile
```

**Review by fcofdez — APPROVED** [review]
LGTM, I left a comment about the expected `google_oauth_client` version but otherwise it looks good. Sorry again for the delay on this!

**Comment by arteam** [comment]
Thanks Francisco!

**Pull request merged**